### PR TITLE
Fix personalisation bugs

### PR DESCRIPTION
**Pull request by davgur**

### CODE REVIEW

File: src/sagas/helpers/url.js [hunk 1]
```diff
@@ -13,7 +13,9 @@ export function* getQuery() {
 
 export function* updateQuery(updater) {
   const query = yield* getQuery();
-  yield put(replace({ search: stringify(updater(query)) }));
+
+  const { state, hash } = (typeof window !== 'undefined') && window.location;
```
Comment (bbfsdev) [inline src/sagas/helpers/url.js:17]: why not           window !== undefined            ?

File: src/redux/modules/auth.js [hunk 2]
```diff
@@ -12,7 +12,7 @@ const UPDATE_TOKEN   = 'Auth/UPDATE_TOKEN';
 export const types = { LOGIN, LOGIN_SUCCESS, LOGOUT_SUCCESS, LOGOUT };
 
 /* Reducer */
-const initialState = { user: null, token: null, error: null, isKcReady: false };
```
Comment (bbfsdev) [inline src/redux/modules/auth.js:15]: why did you remove user from here?

You want to check if it is undefined? vs null?

File: src/components/shared/InitKC.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+const InitKC = ({ children }) => {
+  const dispatch = useDispatch();
+  const language = useSelector(state => settings.getLanguage(state.settings));
+  const user     = useSelector(state => selectors.getUser(state.auth));
+  useEffect(() => {
+    !user && initKC(dispatch, language);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  console.log('InitKC duble bug render', user, user !== undefined);
```
Comment (bbfsdev) [inline src/components/shared/InitKC.js:19]: Did you forget to remove this line?

**Review by bbfsdev — CHANGES_REQUESTED** [review]

File: src/redux/modules/auth.js (hunk 2 above)
Comment (davgur) [reply src/redux/modules/auth.js:15]: yes I need case that user is undefined

**Review by bbfsdev — APPROVED** [review]
